blocks/gu_spoverview/block_gu_spoverview.php

```diff
@@ -45,92 +48,134 @@ public function has_config() {
+        list($inactivities, $aparams) = $DB->get_in_or_equal($activities, SQL_PARAMS_NAMED);
+        list($incourses, $cparams) = $DB->get_in_or_equal($courseids, SQL_PARAMS_NAMED);
+
+        $params = array();
+        $params += $aparams;
+        $params += $cparams;
+
+        if($multipleassessment){
+            $sql = "SELECT DISTINCT i.itemname
+                    FROM `mdl_course_modules` cm 
+                    INNER JOIN `mdl_modules` m ON m.id = cm.module 
+                    INNER JOIN `mdl_grade_items` i on i.courseid = cm.course 
+                    INNER JOIN `mdl_grade_grades` g on g.itemid = i.id 
+                    WHERE m.name {$inactivities} 
+                    AND itemtype = 'mod' 
+                    AND g.finalgrade > 0  
+                    AND g.finalgrade IS NOT NULL 
+                    AND g.userid = $userid
+                    AND cm.course {$incourses}";
+        }
+        else {
+            $sql = "SELECT DISTINCT mcm.id, mcm.course, mcm.instance, mm.name
+                    FROM `mdl_course_modules` mcm
+                    JOIN `mdl_modules` mm ON mm.id = mcm.module
+                    WHERE mm.name {$inactivities}
+                    AND mcm.course {$incourses}";
+        }
+
+        if ($assessments = $DB->get_records_sql($sql, $params)){
+            return $assessments;
+        } else {
+            return $assessments;
+        }
+    }
+
+    public static function retrieve_submitted_assignment($userid){
+        global $DB;
+
+        $submittedassignment = array();
+
+        $sql = "SELECT * FROM `mdl_assign_submission` sub
+                INNER JOIN 	`mdl_assign` ass ON ass.id = sub.assignment 
+                AND userid = $userid AND status='submitted'";
+        
+        if ($submittedassignment = $DB->get_records_sql($sql)){
+            return $submittedassignment;
+        } else {
+            return $submittedassignment;
+        }
+    }
+
+
+    public static function retrieve_assignment_duedate($userid){
+        global $DB;
+
+        $assigmnmentduedate = array();
+
+        $sql = "SELECT DISTINCT i.itemname
+                FROM `mdl_course_modules` cm 
+                INNER JOIN `mdl_modules` m ON m.id = cm.module 
+                INNER JOIN `mdl_grade_items` i ON i.courseid = cm.course 
+                INNER JOIN `mdl_grade_grades` g ON g.itemid = i.id
+                INNER JOIN `mdl_user` u ON u.id = g.userid
+                INNER JOIN 	`mdl_assign` ass ON ass.name = i.itemname
+                INNER JOIN `mdl_grade_grades` gg ON gg.userid = $userid
+                INNER JOIN `mdl_assign_submission` asub ON (asub.assignment = ass.id 
+                AND asub.status <> 'submitted')  
+                WHERE m.name IN ('assign') 
+                AND i.itemmodule IS NOT NULL AND u.id = $userid
+                AND FROM_UNIXTIME(ass.duedate) < NOW()
+                AND gg.finalgrade IS NULL";
+
+        if ($assigmnmentduedate = $DB->get_records_sql($sql)){
```

Comment:
Use Moodle's `$DB->count_records()` or `$DB->count_records_sql()`  instead of using `$DB->get_records_sql()`.